### PR TITLE
Fix window build

### PR DESCRIPTION
**Pull request by achoum**

### CODE REVIEW

File: tools/test_bazel.bat [hunk 1]
```diff
@@ -1,43 +1,51 @@
...
 
 ) ELSE (
 
   copy /Y WORKSPACE_WITH_TF WORKSPACE
 
-  # Without TensorFlow IO.
+  :: Support TensorFlow dataset, but does not use TensorFlow for file IO.
   set FLAGS=--config=windows_cpp17
   %BAZEL% build %FLAGS% //yggdrasil_decision_forests/cli/...:all || goto :error
   %BAZEL% test %FLAGS% //yggdrasil_decision_forests/{cli,metric,model,serving,utils}/...:all //examples:beginner_cc || goto :error
```
Comment (rstz) [inline tools/test_bazel.bat:46]: I'm surprised the tests run, under linux most of our tests require tensorflow

File: yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/BUILD [hunk 2]
```diff
@@ -9,12 +9,12 @@ package(
 # =======
 
 cc_library_ydf(
-    name = "distributed_gradient_boosted_trees",
+    name = "dgbt",
```
Comment (rstz) [inline yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/BUILD:12]: The long path issue is really annoying. This will also require a change in tensorflow_decision_forests/tensorflow/BUILD and possibly parts of the documentation, but we can do this later I think

File: yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/worker.cc [hunk 3]
```diff
@@ -1110,7 +1110,7 @@ DistributedGradientBoostedTreesWorker::EvaluateWeakModelOnvalidationDataset() {
   // Schedule the prediction updates.
   utils::concurrency::StreamProcessor<int, int> processor(
       "update predictions", num_threads,
-      [num_examples, this, num_prediction_dimensions, &caches](
+      [num_examples, this, num_prediction_dimensions, &caches, batch_size](
```
Comment (rstz) [inline yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/worker.cc:1113]: why this change?

File: yggdrasil_decision_forests/utils/test.cc [hunk 4]
```diff
@@ -41,6 +41,12 @@ std::string TmpDirectory() {
   return path;
 }
 
+#ifdef _WIN32
+
+bool IsPortAvailable(int port) {return true;}
```
Comment (rstz) [inline yggdrasil_decision_forests/utils/test.cc:46]: That's surprising, can you explain why there's always a port under windows? Or is this a form of bailing out of the test?

File: yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/worker.cc (hunk 3 above)
Comment (achoum) [reply yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/worker.cc:1113]: VS19 does not automatically inject consts.

File: yggdrasil_decision_forests/utils/test.cc (hunk 4 above)
Comment (achoum) [reply yggdrasil_decision_forests/utils/test.cc:46]: This is a bailing out :).
Let me add a warning.

File: tools/test_bazel.bat (hunk 1 above)
Comment (achoum) [reply tools/test_bazel.bat:46]: We need TensorFlow Proto Example (which is available in this configuration).
What we don't use in this configuration is the TensorFlow IO / Filesystem api. Instead, we use the std c++ one e.g. "std::fopen".

File: yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/BUILD (hunk 2 above)
Comment (achoum) [reply yggdrasil_decision_forests/learner/distributed_gradient_boosted_trees/BUILD:12]: Adding todo

**Comment by rstz** [comment]
Will you do  the copybara import?
